airqo_monitor/get_malfunctions.py

```diff
@@ -31,60 +36,26 @@ def _get_channel_malfunctions(channel_data, channel_type):
         - "no_data": The channel_data list was empty.
     """
     malfunction_list = []
-    if len(channel_data) == 0:
+    if channel_type == AIRQO_CHANNEL_TYPE:
+        malfunction_detector = AirqoMalfunctionDetector()
+    elif channel_type == SOIL_CHANNEL_TYPE:
+        malfunction_detector = SoilMalfunctionDetector()
+    else:
+        malfunction_detector = MalfunctionDetector()
+
+    if malfunction_detector._has_no_data(channel_data):
         malfunction_list.append("no_data")
     else:
-        # TODO(Rachel) - refactor malfunctiond detecting logic to have
-        # fewer individual if cases.
-        if _has_low_battery(channel_data, channel_type):
+        if malfunction_detector._has_low_battery(channel_data):
```

Comment:
i was imagining each detector can just run its own set of tests instead of being limited to the ones here. e.g. it has a public facing function get_malfunctions() which runs the list of detectors and returns the list of malfunctions.